internal/datasystem/store.go

```diff
@@ -300,6 +301,32 @@ func (s *Store) getActive() subsystems.ReadOnlyStore {
 	return s.active
 }
 
+//nolint:revive // Implementation for ReadOnlyDataStore.
+func (s *Store) Snapshot() (
```

Comment:
When the relay tries to grab information from the store to determine what it should play back, it needs to get all the store information along with the selector in consistent fashion. Doing so increases data consistency but also will make some implementation details easier in the relay which you will see in the related PR [here](https://github.com/launchdarkly/ld-relay/pull/501).